veps/sig-compute/55-evictionstrategy-extension.md

```diff
@@ -0,0 +1,154 @@
+# VEP #55: Extend `EvictionStrategy` to Support Specific Cluster-Initiated `virt-launcher` Pod Disruptions
+
+## Release Signoff Checklist
+
+Items marked with (R) are required *prior to targeting to a milestone / release*.
+
+- [X] (R) Enhancement issue created, which links to VEP dir in [kubevirt/enhancements] (not the initial VEP PR)
+- [ ] (R) Target version is explicitly mentioned and approved
+- [X] (R) Graduation criteria filled
+
+## Overview
+
+Under specific well-known circumstances (e.g., scheduler preemption, `NoExecute` taints), a `virt-launcher` pod can be terminated (read deleted).
+In these same cases, `PodDisruptionBudget` can also be bypassed. This can easily result in a VM being gracefully shut down, even if its `EvictionStrategy` is set to `LiveMigrate` or `LiveMigrateIfPossible`.
+
+This proposal aims to have our controllers identify well-known cases (initially scheduler preemption and `NoExecute` taints) and, if the `EvictionStrategy` requires `LiveMigrate` or `LiveMigrateIfPossible`, attempt a live migration.
```

Comment:
What happens if live migration fails?
What will be the fallback?

Reply:
Yes, this is the real question but in my opinion we already have this problem today.
Let's assume that a migration of a specific VM cannot complete for any possible reason.
Even when using the eviction API, today we do not have any fallback logic:
the VMI will remain marked for eviction and the eviction controller will try migrations forever.
On my opinion we already need a fallback logic now independently of this.